pkg/manifests/updates.go

```diff
@@ -100,25 +96,22 @@ func UpdateSchedulerConfigNamespaces(logger tlog.Logger, cm *corev1.ConfigMap, N
 	binData, err := KubeSchedulerConfigurationToData(kc)
 	if err != nil {
 		logger.Debugf("cannot encode the KubeSchedulerConfiguration: %v", err)
-		return cm
+		return
 	}
 	cm.Data["scheduler-config.yaml"] = string(binData)
-	return cm
 }
 
-func UpdateResourceTopologyExporterDaemonSet(plat platform.Platform, ds *appsv1.DaemonSet, cm *corev1.ConfigMap, pullIfNotPresent bool, nodeSelector *metav1.LabelSelector) *appsv1.DaemonSet {
+func UpdateResourceTopologyExporterDaemonSet(plat platform.Platform, ds *appsv1.DaemonSet, cm *corev1.ConfigMap, pullIfNotPresent bool, nodeSelector *metav1.LabelSelector) {
 	// TODO: better match by name than assume container#0 is RTE proper (not minion)
 	ds.Spec.Template.Spec.Containers[0].Image = images.ResourceTopologyExporterImage
 	ds.Spec.Template.Spec.Containers[0].ImagePullPolicy = pullPolicy(pullIfNotPresent)
 	if len(ds.Spec.Template.Spec.Containers) > 1 {
 		// TODO: more polite/proper iteration
 		ds.Spec.Template.Spec.Containers[1].ImagePullPolicy = pullPolicy(pullIfNotPresent)
 	}
-	vars := map[string]string{
-		"RTE_POLL_INTERVAL": "10s",
-		"EXPORT_NAMESPACE":  ds.Namespace,
-	}
-	ds.Spec.Template.Spec.Containers[0].Command = UpdateResourceTopologyExporterCommand(ds.Spec.Template.Spec.Containers[0].Command, vars, plat)
+
```

Comment:
Can we move part of the command line configuration under `GetManifests` in general I am expecting that `UpdateResourceTopologyExporterDaemonSet` should update the only dynamical configuration? If each time it will apply the same arguments we definitely can move it under the `GetManifests` underlying methods.
Same true regarding `plat` parameter, like it no taste to pass it to the Update method when you should know the platform from the beginning and it should stay the same during the run.

Reply:
that's a good point. In general this is indeed the gist. There are a lot of nuances and poorly-specified areas, so let's gradually cleanup.

Reply:
initial change: https://github.com/k8stopologyawareschedwg/deployer/pull/60/commits/e9ba6fa5d34329446d40e842ae3751efac60a8a3

we will probably have more in the future